openfeature/provider/provider.py

```diff
@@ -66,3 +67,22 @@ def resolve_object_details(
         evaluation_context: typing.Optional[EvaluationContext] = None,
     ) -> FlagResolutionDetails[typing.Union[dict, list]]:
         pass
+
+    def emit_provider_ready(self, details: ProviderEventDetails) -> None:
```

Comment:
The ability to manually emit this and other events is needed.

In addition, the SDK should automatically emit READY/ERROR based on the suceess/failure of the init function.

[5.3.1](https://openfeature.dev/specification/sections/events/#requirement-531), [5.3.2](https://openfeature.dev/specification/sections/events/#requirement-532). Again there's a bit more info in the unreleased specification: https://github.com/open-feature/spec/blob/31da456d953752522c557ee0b7cce7db7bba331e/specification/sections/05-events.md?plain=1#L130

Reply:
Been thinking about how to approach these requirements and I think it's best to first implement provider status in a separate pull request and then integrate those changes here.